===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(p11_rpc_buffer_new_full:buffer)) at 6 (line 529, column 7) }
===== Actual patch in diff format =====
@@ -527,6 +527,7 @@ p11_rpc_buffer_new_full (size_t reserve,
 
 	p11_buffer_init_full (buffer, NULL, 0, 0, frealloc, ffree);
 	if (!p11_buffer_reset (buffer, reserve))
+		free(buffer);
 		return_val_if_reached (NULL);
 
 	return buffer;
